webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/score/ScoreSerializer.scala

```diff
@@ -0,0 +1,114 @@
+package com.normation.rudder.score
+
+import com.normation.errors._
+import com.normation.errors.IOResult
+import com.normation.rudder.domain.reports.ComplianceSerializable
+import com.normation.zio._
+import zio.Ref
+import zio.ZIO
+import zio.json._
+import zio.json.ast.Json
+import zio.syntax._
+
+object ScoreSerializer {
+
+  implicit val scoreValueEncoder: JsonEncoder[ScoreValue] = JsonEncoder[String].contramap(_.value)
+  implicit val scoreValueDecoder: JsonDecoder[ScoreValue] = JsonDecoder[String].mapOrFail(ScoreValue.fromString)
+
+  implicit val noDetailsScoreEncoder: JsonEncoder[NoDetailsScore] = DeriveJsonEncoder.gen
+  implicit val noDetailsScoreDecoder: JsonDecoder[NoDetailsScore] = DeriveJsonDecoder.gen
+
+  implicit val globalScoreEncoder: JsonEncoder[GlobalScore] = DeriveJsonEncoder.gen
+  implicit val globalScoreDecoder: JsonDecoder[GlobalScore] = DeriveJsonDecoder.gen
+
+  implicit val jsonScoreEncoder: JsonEncoder[JsonScore] = DeriveJsonEncoder.gen
+  implicit val jsonScoreDecoder: JsonDecoder[JsonScore] = DeriveJsonDecoder.gen
+}
+
+trait ScoreSerializerService {
+  def parse(score:  Score[Json]): IOResult[Option[Score[_]]]
```

Comment:
I think it should be a pure result, I don't see why we should have IO here

Reply:
maybe i need some services that may not be pure

Reply:
I didn't see any and parsing should really always be pure (and perhaps you need some impure one before geting the data, but having IO involved in the parsing feels really like not what we want to have to deal with)

Reply:
Also it's uised mostly in ZIO context and i have to transform my PureResult into IORestult to i guess it does not gain much

Reply:
it gains in intent conveyed, and in unit test easyness. `IOResult` is a bit like global variables, they are easy and powerfull, but we should always try to restraint as much as possible to use them because they are *too* powerful.

Reply:
(not a subject anymore since that part of the code disapeared, it's for info)